### PR TITLE
Setup OpenAlex frontfill to run every night

### PR DESCRIPTION
**Pull request by manveerxyz**
- Setup new `pull_new_openalex_papers` to run every night
- Save logs in db
- Update `celery_process_paper` to also track `citations` and `open_alex_raw_json`

### CODE REVIEW

File: src/paper/tasks.py [hunk 1]
```diff
@@ -1267,3 +1267,157 @@ def pull_openalex_author_works(user_id, openalex_id):
         filters=[NEW],
     )
     return True
+
+
+@periodic_task(
+    run_every=crontab(minute=0, hour=0), priority=3, queue=QUEUE_PULL_PAPERS
```
Comment (manveerxyz) [inline src/paper/tasks.py:1273]: I might change this from midnight to sometime in the early morning. Going to take a look at avg. db load by time of day to get a sense of when would be best.

Comment (yattias) [reply src/paper/tasks.py:1273]: Great thinking

File: src/paper/related_models/paper_model.py [hunk 2]
```diff
@@ -967,6 +967,33 @@ def get_url_method_priority_list(cls, url):
             # methods.append(cls.MANUBOT_PDF_URL)
             methods.append(cls.MANUBOT_URL)
         return methods
+    
+
+class PaperFetchLog(models.Model):
```
Comment (yattias) [inline src/paper/related_models/paper_model.py:972]: Nice, this is good to have

File: src/paper/tasks.py [hunk 3]
```diff
@@ -1267,3 +1267,157 @@ def pull_openalex_author_works(user_id, openalex_id):
...
+            fetch_type=PaperFetchLog.FETCH_NEW,
+            status=PaperFetchLog.SUCCESS,
+            started_date=start_date,
+            completed_date=datetime.now(),
+            num_papers_success=num_papers_success,
+            num_papers_failed=num_papers_failed,
+        )
+    except requests.ConnectionError:
+        pull_new_openalex_works.apply_async(
+            (page, retry + 1), priority=4, countdown=10 + (retry * 2)
```
Comment (yattias) [inline src/paper/tasks.py:1412]: NIce

**Review by yattias — APPROVED** [review]

**Pull request merged**